alert-database/src/test/java/com/synopsys/integration/alert/database/api/mock/MockProviderTaskPropertiesRepository.java

```diff
@@ -0,0 +1,32 @@
+package com.synopsys.integration.alert.database.api.mock;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import com.synopsys.integration.alert.database.provider.task.ProviderTaskPropertiesEntity;
+import com.synopsys.integration.alert.database.provider.task.ProviderTaskPropertiesRepository;
+
+public class MockProviderTaskPropertiesRepository extends DefaultMockJPARepository<ProviderTaskPropertiesEntity, Long> implements ProviderTaskPropertiesRepository {
+
+    private Map<Long, ProviderTaskPropertiesEntity> providerTaskPropertiesEntityMap = new HashMap<>();
+
+    public MockProviderTaskPropertiesRepository(Long providerConfigId, String taskName, String propertyName, String value) {
+        ProviderTaskPropertiesEntity providerTaskPropertiesEntity = new ProviderTaskPropertiesEntity(providerConfigId, taskName, propertyName, value);
+        providerTaskPropertiesEntityMap.put(providerConfigId, providerTaskPropertiesEntity);
```

Comment:
Should call the save method here, in case we need to add additional behavior for saving entities in this Mock class in the future.